### PR TITLE
fix(cognito): allow custom email msg placeholder

### PR DESCRIPTION
**Pull request by GavinZZ**
When I change the text in the "{##Verify Email##}" placeholder, e.g. to "{##verify your email##}", cdk synth and cdk deploy commands yield the error,
```
Error: Verification email body must contain the template string '{##Verify Email##}'
```
This is incorrect as that documentation states that this string may be customised.
https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-email-verification-message-customization.html

Help text in the AWS Cognito console reads:

    You can customize this message with HTML. "Verify email" is the text that will be displayed over the clickable link in the message. You can customize the "Verify email" string, but the variable - some text enclosed by "{##" and "##}" - must be kept in the message.

Expected Behavior

Be able to change the emailBody property of a UserPool to include the placeholder of format, "{##Verify Your Email##}" with any custom string allowed by AWS Cognito.

Closes https://github.com/aws/aws-cdk/issues/23828

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts [hunk 1]
```diff
@@ -1069,7 +1070,7 @@
     } else {
       const emailMessage = props.userVerification?.emailBody ??
         `Verify your account by clicking on ${VERIFY_EMAIL_TEMPLATE}`;
-      if (!Token.isUnresolved(emailMessage) && emailMessage.indexOf(VERIFY_EMAIL_TEMPLATE) < 0) {
+      if (!Token.isUnresolved(emailMessage) && !VERIFY_EMAIL_REGEX.test(emailMessage)) {
```
[the remark this answers is not in the document]
Comment (GavinZZ) [reply packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts:1073]: Fixed using lazy qualifier `?`, which will return for the fist `##` found.

File: packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-link-custom-placeholder.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  },
+});
+
+new CfnOutput(stack, 'user-pool-id', {
+  value: userpool.userPoolId,
+});
+
+new IntegTest(app, 'user-pool-link-custom-placeholder-integ-test', {
+  testCases: [stack],
+  cdkCommandOptions: {
```
Comment (ConnorRobertson) [inline packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-link-custom-placeholder.ts:25]: What's this?

Comment (GavinZZ) [reply packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-link-custom-placeholder.ts:25]: As far as I understand, it enables options to certain CDK CLI commands. In this case, it allows the `cdk destroy` command to forcefully delete the user pool.

File: packages/aws-cdk-lib/aws-cognito/test/user-pool.test.ts [hunk 3]
```diff
@@ -161,14 +161,35 @@ describe('User Pool', () => {
...
+      userVerification: {
+        emailStyle: VerificationEmailStyle.LINK,
+        emailBody: 'invalid email body {##Verify !! Email##}',
+      },
+    })).toThrow(/Verification email body/);
+
+    expect(() => new UserPool(stack, 'Pool9', {
+      userVerification: {
+        emailStyle: VerificationEmailStyle.LINK,
+        emailBody: 'invalid email body {##Click here to verify##}',
```
Comment (gracelu0) [inline packages/aws-cdk-lib/aws-cognito/test/user-pool.test.ts:190]: this `emailBody` is valid right? if so it's a bit confusing that it says 'invalid email body'

Comment (GavinZZ) [reply packages/aws-cdk-lib/aws-cognito/test/user-pool.test.ts]: good catch.

File: packages/aws-cdk-lib/aws-cognito/test/user-pool.test.ts [hunk 4]
```diff
@@ -159,14 +159,35 @@ describe('User Pool', () => {
...
+      userVerification: {
+        emailStyle: VerificationEmailStyle.LINK,
+        emailBody: 'invalid email body ##Verify Email##',
+      },
+    })).toThrow(/Verification email body/);
+
+    expect(() => new UserPool(stack, 'Pool8', {
+      userVerification: {
+        emailStyle: VerificationEmailStyle.LINK,
+        emailBody: 'invalid email body {##Verify !! Email##}',
```
Comment (gracelu0) [inline packages/aws-cdk-lib/aws-cognito/test/user-pool.test.ts:183]: from the [docs](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-email-verification-message-customization.html), it says that `The maximum length for the message, including the verification code (if present), is 20,000 UTF-8 characters. You can use HTML tags in this message to format the contents.`. so I believe some special characters (like `!`) are allowed? if so we might want to update the regex accordingly

Comment (GavinZZ) [reply packages/aws-cdk-lib/aws-cognito/test/user-pool.test.ts:183]: good catch, looking at the [cfn documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cognito-userpool-verificationmessagetemplate.html#cfn-cognito-userpool-verificationmessagetemplate-emailmessagebylink), the regex they used is `\{##[\p{L}\p{M}\p{S}\p{N}\p{P}\s*]*##\}` which includes symbol, punctuation and accent characters. Let me update it.

**Review by gracelu0 — APPROVED** [review]
thanks for making the changes! lgtm :)

**Pull request merged**